code/modules/atmospherics/machinery/atmosmachinery.dm

```diff
@@ -169,7 +169,7 @@
 
 	for(var/i in 1 to device_type)
 		for(var/obj/machinery/atmospherics/target in get_step(src,node_connects[i]))
-			if(can_be_node(target, i))
+			if(can_be_node(target) && target.can_be_node(src))
```

Comment:
What is this trying to fix exactly? if is the machines not anchored i believe is better to add an anchored check here or down the line, mostly because there is already a target call for isConnectable down the line

Reply:
before change:
A pasess node check to B, B doesn't pass node check to A
A gets connected to B
B is not connected to A
then the system will assume wrong variables and other stuff and runtimes runtimes runtimes bugs
after change:
A passes node check to B, but B doesn't pass it to A, they dont connect and all is fine

Depending on your circumstances and design an unanchored machine could still be a valid target for node connectivity, it's why the proc exists. Using can_be_node() you can check for all the nessecary stuff you come up with, if you rely on an anchored check you end up being very limited

Reply:
but yes I wasn't aware isConnectable() is a thing, I saw a use of targeted check through can_be_node(). Yeah so I assume TEG uses it improperly and it gave me a wrong example

Reply:
yeah I'll remove that change